pytorch_tabnet/abstract_model.py

```diff
@@ -1,134 +1,90 @@
+    scheduler_fn: Any = None
+    scheduler_params: Dict = field(default_factory=dict)
+    mask_type: str = "sparsemax"
+    input_dim: int = None
+    output_dim: int = None
+    device_name: str = "auto"
+
+    def __post_init__(self):
         self.batch_size = 1024
-
-        self.seed = seed
+        self.virtual_batch_size = 1024
         torch.manual_seed(self.seed)
         # Defining device
-        if device_name == 'auto':
+        if self.device_name == "auto":
             if torch.cuda.is_available():
-                device_name = 'cuda'
+                device_name = "cuda"
             else:
-                device_name = 'cpu'
+                device_name = "cpu"
         self.device = torch.device(device_name)
         print(f"Device used : {self.device}")
 
-    @abstractmethod
-    def construct_loaders(self, X_train, y_train, X_valid, y_valid,
-                          weights, batch_size, num_workers, drop_last):
-        """
-        Returns
-        -------
-        train_dataloader, valid_dataloader : torch.DataLoader, torch.DataLoader
-            Training and validation dataloaders
-        -------
-        """
-        raise NotImplementedError('users must define construct_loaders to use this base class')
-
-    def init_network(
-                     self,
-                     input_dim,
-                     output_dim,
-                     n_d,
-                     n_a,
-                     n_steps,
-                     gamma,
-                     cat_idxs,
-                     cat_dims,
-                     cat_emb_dim,
-                     n_independent,
-                     n_shared,
-                     epsilon,
-                     virtual_batch_size,
-                     momentum,
-                     device_name,
-                     mask_type,
-                     ):
-        self.network = tab_network.TabNet(
-            input_dim,
-            output_dim,
-            n_d=n_d,
-            n_a=n_a,
-            n_steps=n_steps,
-            gamma=gamma,
-            cat_idxs=cat_idxs,
-            cat_dims=cat_dims,
-            cat_emb_dim=cat_emb_dim,
-            n_independent=n_independent,
-            n_shared=n_shared,
-            epsilon=epsilon,
-            virtual_batch_size=virtual_batch_size,
-            momentum=momentum,
-            device_name=device_name,
-            mask_type=mask_type).to(self.device)
-
-        self.reducing_matrix = create_explain_matrix(
-            self.network.input_dim,
-            self.network.cat_emb_dim,
-            self.network.cat_idxs,
-            self.network.post_embed_dim)
-
-    def fit(self, X_train, y_train, X_valid=None, y_valid=None, loss_fn=None,
-            weights=0, max_epochs=100, patience=10, batch_size=1024,
-            virtual_batch_size=128, num_workers=0, drop_last=False):
+    def fit(
+        self,
+        X_train,
+        y_train,
+        eval_set=None,
```

Comment:
default eval_set should be `[]` otherwise you get an error when not giving `eval_set`

Reply:
done